### PR TITLE
added template

### PR DESCRIPTION
**Pull request by maikvandergaag**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-webapp-including-insights/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 101-webapp-including-insights/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    "webappName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: add a defaultValue w/ uniquestring

File: 101-webapp-including-insights/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        "description": "The name of the web application"
+      }
+    },
+    "appInsightsName": {
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: move to a variable

File: 101-webapp-including-insights/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "The name of the application insights instance"
+      }
+    },
+    "hostingPlanName": {
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: add a defaultValue w/ uniquestring

File: 101-webapp-including-insights/azuredeploy.json (hunk 3 above)
Comment (maikvandergaag) [reply 101-webapp-including-insights/azuredeploy.json]: done

File: 101-webapp-including-insights/azuredeploy.json (hunk 1 above)
Comment (maikvandergaag) [reply 101-webapp-including-insights/azuredeploy.json]: done

File: 101-webapp-including-insights/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,142 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "webappName": {
+      "type": "string",
+      "defaultValue": "GEN-UNIQUE",
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: this needs to be to use the uniqueString() function

File: 101-webapp-including-insights/azuredeploy.json (hunk 3 above)
Comment (bmoore-msft) [reply 101-webapp-including-insights/azuredeploy.json]: doesn't look like it was moved to a variable

File: 101-webapp-including-insights/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      "minLength": 1,
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "The location for the resources"
+      }
+    },
+   "hostingPlanName": { 
+      "type": "string",
+      "minLength": 1,
+      "defaultValue": "GEN-UNIQUE",
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: need to use uniqueString()

File: 101-webapp-including-insights/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    "skuName": {
+      "type": "string",
+      "defaultValue": "F1",
+      "metadata": {
+        "description": "The hosting plan's pricing tier and capacity. Check details at https://azure.microsoft.com/en-us/pricing/details/app-service/"
+      }
+    },
+    "owner": {
+      "type": "string",
+      "defaultValue": "GEN-UNIQUE",
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: this doesn't need to be unique so you can use a literal value here

File: 101-webapp-including-insights/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+        "[resourceId('Microsoft.Web/serverfarms/', parameters('hostingPlanName'))]",
+        "[resourceId('microsoft.insights/components/', parameters('appInsightsName'))]"
+      ],
+      "properties": {
+        "name": "[parameters('webappName')]",
+        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', parameters('hostingPlanName'))]"
+      },
+      "resources": [
+        {
+          "apiVersion": "2015-08-01",
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: update apiVersion

File: 101-webapp-including-insights/azuredeploy.parameters.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    "webappName": {
+      "value": "GEN-UNIQUE"
+    },
+    "appInsightsName": {
+      "value": "GEN-UNIQUE"
+    },
+    "hostingPlanName": {
+      "value": "GEN-UNIQUE"
+    },
+    "owner": {
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.parameters.json]: remove this since you'll have a defaultValue

Comment (maikvandergaag) [reply 101-webapp-including-insights/azuredeploy.parameters.json]: done

File: 101-webapp-including-insights/azuredeploy.json (hunk 8 above)
Comment (maikvandergaag) [reply 101-webapp-including-insights/azuredeploy.json]: done

File: 101-webapp-including-insights/azuredeploy.json (hunk 7 above)
Comment (maikvandergaag) [reply 101-webapp-including-insights/azuredeploy.json]: done

File: 101-webapp-including-insights/azuredeploy.json (hunk 6 above)
Comment (maikvandergaag) [reply 101-webapp-including-insights/azuredeploy.json]: done

File: 101-webapp-including-insights/azuredeploy.json (hunk 5 above)
Comment (maikvandergaag) [reply 101-webapp-including-insights/azuredeploy.json]: done

**Comment by maikvandergaag** [comment]
I didn't move the "appInsightsName" parameter to a variable because I think users should be able to fill it in especially when using specific naming conventions.

**Comment by maikvandergaag** [comment]
@bmoore-msft think everything is correct build is failing for:  Template Local Validation 101-webapp-including-insights metadata.json matches schema:   AssertionError: #:Missing required property "$schema". Is this a new requirement within the metadata.json file? can not find anything in the contribution guide.

File: 101-webapp-including-insights/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,142 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "webappName": {
+      "type": "string",
+      "defaultValue": "uniqueString()",
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: uniqueString needs to be seeded with something

"defaultValue": "[concat('web-', uniqueString(resourceGroup().id))]"

File: 101-webapp-including-insights/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+      "minLength": 1,
+      "defaultValue": "uniqueString()",
+      "metadata": {
+        "description": "The name of the application insights instance"
+      }
+    },
+    "location": {
+      "type": "string",
+      "minLength": 1,
+      "defaultValue": "westeurope",
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: use [resourceGroup().location] for the defaultValue

File: 101-webapp-including-insights/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+      "metadata": {
+        "description": "The location for the resources"
+      }
+    },
+    "hostingPlanName": {
+      "type": "string",
+      "minLength": 1,
+      "defaultValue": "uniqueString()",
```
Comment (bmoore-msft) [inline 101-webapp-including-insights/azuredeploy.json]: same as above

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**